neuralop/training/trainer.py

```diff
@@ -790,3 +790,47 @@ def checkpoint(self, save_dir):
                 print(f"[Rank 0]: saved training state to {save_dir}")
 
        
+
+class PINOTrainer(Trainer):
+    """
+    Trainer subclass that logs per-field loss values and weights at the end of each epoch.
+    Works with meta-losses like Relobralo that expose latest_loss_record, latest_weight_record, and latest_total_loss attributes.
+    """
+    def train_one_epoch(self, epoch, train_loader, training_loss):
```

Comment:
Like mentioned before - might it be easier to modify `PINOTrainer.train_one_batch` as well/instead of `train_one_epoch`? That way you'd have more control over the loss fn's signature and you wouldn't need to assume a fully abstract loss.